assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/clip/mlflow_wrapper.py

```diff
@@ -0,0 +1,136 @@
+
+import mlflow
+from PIL import Image
+import pandas as pd
+import torch
+import tempfile
+
+from transformers import CLIPProcessor, CLIPModel
+from config import MLflowSchemaLiterals, MLflowLiterals, Tasks
+from utils import create_temp_file, process_image, get_current_device
+from typing import List, Tuple
+
+
+class CLIPMLFlowModelWrapper(mlflow.pyfunc.PythonModel):
+    """MLflow model wrapper for CLIP model."""
+
+    def __init__(
+        self,
+        task_type: str,
+    ) -> None:
+        """Constructor for MLflow wrapper class
+        :param task_type: Task type used in training.
+        :type task_type: str
+        """
+        super().__init__()
+        self._processor = None
+        self._model = None
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        if self._task_type == Tasks.ZERO_SHOT_IMAGE_CLASSIFICATION.value:
+            try:
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._processor = CLIPProcessor.from_pretrained(model_dir)
+                self._model = CLIPModel.from_pretrained(model_dir)
+                self._device = get_current_device()
+                self._model.to(self._device)
+
+                print("Model loaded successfully")
+            except Exception as e:
+                print("Failed to load the the model.")
+                print(e)
+                raise
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+        :param input_data: Input images for prediction.
+        :type input_data: Pandas DataFrame with a first column name ["image"] of images where each
+        image is in base64 String format.
+        :param task: Task type of the model.
+        :type task: HFTaskLiterals
+        :param tokenizer: Preprocessing configuration loader.
+        :type tokenizer: transformers.AutoImageProcessor
+        :param model: Pytorch model weights.
+        :type model: transformers.AutoModelForImageClassification
+        :return: Output of inferencing
+        :rtype: Pandas DataFrame with columns ["filename", "probs", "labels"] for classification and
+        ["filename", "boxes"] for object detection, instance segmentation
+        """
+        # Decode the base64 image column
+        decoded_images = input_data.loc[
+            :, [MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]
+        ].apply(axis=1, func=process_image)
+
+        captions = input_data[MLflowSchemaLiterals.INPUT_COLUMN_TEXT].iloc[0].split(',')
```

Comment:
what if captions are different for 2nd example in request?

Reply:
The model is only able to take in one set of labels, so we expect that the user passes in the labels as the first row of the text column and provides empty strings for the rest of the rows. I have added validation and updated documentation for this.